wordpress-mobile/WordPress-Android
# IA reader accessibility audit

Partially fixes #10962 (the items related to Reader directly)

**This PR is based on #11270 that must be merged before this one can be merged (currently targeting that PR branch to ease reviewing Files changed); UI design changes already approved there, so this Accessibility PR can be reviewed even if not merged before**

We made an audit of the Reader Subfilter component and Bottom Sheet using `Accessibility Scanner` and `TalkBack`:

| Subfilter Component | Reader Bottom Sheet |
|---|---|
| ![image](https://user-images.githubusercontent.com/47797566/74727111-efa25800-5240-11ea-8394-8ec885414089.png) | ![image](https://user-images.githubusercontent.com/47797566/74726977-b0740700-5240-11ea-8982-622518f4116a.png) |

## What we changed based on Accessibility Scanner Audit and TalkBack testing
1. Subfilter Component (`subfilter_component.xml`)
   a. Added contentDescription to relevant views and declaring `android:id="@+id/filter_list_button"` as not important for accessibility
   b. Increased the `android:id="@+id/remove_filter_button"` width to 48dp
2. Reader Bottom Sheet:
   a. Created focusable group in `subfilter_list_item.xml` and added relevant contentDescription so that the selected item in the bottom sheet is announced as selected from TalkBack
   b. Marked `android:id="@+id/empty_state_container"` as important for accessibility so to get the inner content announced directly

## NOTES
- We did not marked SITES and TAGS bottom sheet tabs as Headings since now they are tabs (in previous design we had a unique list of items)
- When no TAGS/SITES are present (like in the image below) we kept the text button height to the standard one I see we use across the app `@dimen/default_dialog_button_height` whereas we set a minWidth of `@dimen/min_touch_target_sz`
<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/47797566/74728534-3bee9780-5243-11ea-8f94-72b6764fe120.png">
</p>

## To test
Referring to the `What we changed based on Accessibility Scanner Audit and TalkBack testing` section above:
1. Subfilter Component: rerun Accessibility Scanner and check the above reported changes were effective
2. Reader Bottom Sheet: use talkback (tested with default navigation): use TalkBack to navigate the Bottom Sheet and check that the above reported changes were effective.

PR submission checklist:
